tools/src/compileProtos.ts

```diff
@@ -372,36 +373,42 @@ export async function main(parameters: string[]): Promise<void> {
   const protoJsonFiles: string[] = [];
   let skipJson = false;
   let esm = false;
+  let outputDir = '.';
   const directories: string[] = [];
-  for (const parameter of parameters) {
-    if (parameter === '--skip-json') {
+  for (let x = 0; x < parameters.length; x++) {
+    if (parameters[x] === '--skip-json') {
       skipJson = true;
       continue;
     }
-    if (parameter === '--esm') {
+    if (parameters[x] === '--esm') {
       esm = true;
       continue;
     }
+    if (parameters[x] === '--out-dir') {
+      outputDir = parameters[x + 1];
```

Comment:
if you do this, on the next iteration the directory will be also taken as a non-option parameter, and will be added to `directories`.

Shall we just start using `yaqs` at this point?